File: vagrant-k8s/apl-microservice-art/db_setup_tools/create_db.js
#!/usr/bin/env node
/*                                                                                        
  データベース新規作成
*/


// 固有名称
var dbn = "session_express";
var service_instance_name = 'kube-cloudant';
var vcap_path = '../webapp1/config/vcap-local.json';


// 環境からの接続情報取得
var cfenv = require("cfenv");
var vcapLocal;
try {
    vcapLocal = require(vcap_path);
} catch (err) {
    throw err;
}
var appEnvOpts = vcapLocal ? { vcap: vcapLocal} : {}
var appEnv = cfenv.getAppEnv(appEnvOpts);
var svc = appEnv.getServiceCreds(service_instance_name);


// クラウダントとの接続処理
var Cloudant = require('cloudant');
var cloudant = Cloudant(svc.url);


// DB削除                                                                                   
cloudant.db.destroy(dbn, function(err) {
  // DB作成
  cloudant.db.create(dbn, function(err) {
    if (err) throw err;
    console.log("データベース作成成功");
  });
});

File: vagrant-k8s/apl-microservice-art/webapp1/db_setup_tools/create_db.py
#!/usr/bin/env python                                                           
# -*- coding:utf-8 -*-                                                          
#
# データベース新規作成
#

import sys
import json
from cloudant.client import Cloudant

# 固有名称
database_name = 'sample'
label = 'cloudantNoSQLDB'
i = 0
vcap_path = '../webapp1/config/vcap-local.json'

cred = None

# Cloudant認証情報の取得
try:
  f = open(vcap_path, 'r')
  cred = json.load(f)
  f.close()
except:
    print "ERROR==================="
    import traceback
    traceback.print_exc()

json.dumps(cred)

# クラウダントへの接続
client = Cloudant(cred['services'][label][i]['credentials']['username'], 
                  cred['services'][label][i]['credentials']['password'], 
                  url=cred['services'][label][i]['credentials']['url'])
client.connect()

# DBが存在していたら削除
print "既存データベース ", database_name ," の削除"
try:
    db = client[database_name]
    if db.exists():
        client.delete_database(database_name)
except:
    pass


# DBを新規作成
print "新規データベース ", database_name ," の作成"
try: 
    db = client.create_database(database_name)
    print "データベース作成成功"
except:
    print "データベース作成失敗"
    sys.exit()


